### PR TITLE
Add mandatory field for cleanup policy

### PR DESCRIPTION
**Pull request by ferbncode**
Add cleanup policy field for output event type.

### CODE REVIEW

File: client/Pages/EventTypeCreate/Query.elm [hunk 1]
```diff
@@ -58,6 +59,41 @@ viewQueryForm model =
...
+                    ]
+                , if (getValue FieldCleanupPolicy formModel.values) == cleanupPolicies.compact then
+                    textInput formModel
+                        FieldPartitionCompactionKeyField
+                        OnInput
+                        "Partition Compaction Key Field"
+                        "Example: product.order_name"
+                        "Field to be used as partition_compaction_key"
+                        Help.partitionCompactionKeyField
+                        Optional
```
Comment (lmontrieux) [inline client/Pages/EventTypeCreate/Query.elm]: Shouldn't it be required in this case?

File: client/Pages/EventTypeCreate/Query.elm [hunk 2]
```diff
@@ -153,6 +189,7 @@ submitQueryCreate model =
               , Json.object
                     [ ( "name", asString FieldName )
                     , ( "owning_application", asString FieldOwningApplication )
+                    , ( "cleanup_policy", asString FieldCleanupPolicy )
```
Comment (lmontrieux) [inline client/Pages/EventTypeCreate/Query.elm]: Shouldn't you also add partition compaction key fields or retention period, when appropriate?

Comment (ferbncode) [reply client/Pages/EventTypeCreate/Query.elm]: Right! Forgot to do that :sweat_smile:

File: client/Pages/EventTypeCreate/Models.elm [hunk 3]
```diff
@@ -40,6 +40,7 @@ type Field
     | FieldAudience
     | FieldCleanupPolicy
     | FieldSql
+    | FieldPartitionCompactionKeyField
```
Comment (SergKam) [inline client/Pages/EventTypeCreate/Models.elm]: Name contains `Field` twice. `FieldPartitionCompactionKey` is enouth

File: client/Pages/EventTypeDetails/Help.elm [hunk 4]
```diff
@@ -495,3 +495,20 @@ authorization =
...
+     , text " are both log-compacted, then the compaction key of input is used as the compaction key"
+     , text " for the output."
+    , newline
+    , bold "This field is mandatory only if input event type is non log-compacted and output event type is compacted"
+    , newline
+    , newline
+    , bold "Key: "
+    , mono "partion_compaction_key_field"
+    , bold "optional"
+    ]
```
Comment (SergKam) [inline client/Pages/EventTypeDetails/Help.elm]: Add EOF

File: client/Pages/EventTypeCreate/Query.elm (hunk 2 above)
Comment (SergKam) [inline client/Pages/EventTypeCreate/Query.elm]: Don't forget retention time

File: client/Pages/EventTypeCreate/Models.elm (hunk 3 above)
Comment (ferbncode) [reply client/Pages/EventTypeCreate/Models.elm]: We added a new field called `partition_compaction_key_field` via which the user can specify a non-nullable string field to be used as partition_compaction_key for the ouput event type (https://github.bus.zalan.do/aruha/nakadi-sql-api/blob/master/docs/_data/nakadi-sql-api-definition.yaml#L292)

**Pull request merged**